app/controllers/kudos_controller.rb

```diff
@@ -20,10 +20,15 @@ def edit
   def create
     @kudo = Kudo.new(kudo_params)
     @kudo.giver_id = current_employee.id
-    if @kudo.save
-      Employee.transaction do
+    Employee.transaction do
+      if @kudo.save!
```

Comment:
when this save! will fail, nothing below will run, therefore there will be no `render 'new'` action. you need to rescue this error 

you can try to fail this save, by setup some validations and pushing kudo withot this validation

also specs are red

Reply:
https://www.google.com/search?client=firefox-b-d&q=rails+transaction+rescue